### ISSUE TITLE
Multiple namespaces in one class are not bound correctly

### ISSUE THREAD

**Issue by suushiemaniac** [pdvrieze/xmlutil#36]
Kotlin: 1.3.72 JVM
KotlinX serialization runtime: 0.20.0
xmlutil: 0.20.0.1

When annotating any serializable class with a namespaced `XmlSerialName` and then also annotating any of its properties with another, different namespaces, the individual namespaces are not prefixed correctly in the serialized XML.

```kotlin
@Serializable
@XmlSerialName("namespaced", "http://example.org", "ex")
data class PrefixBreaks(
    @XmlSerialName("property", "http://myownnamespace.com", "foo") val property: String
)

val thisExplodes = PrefixBreaks("bug")

val defaultSerializer = XML { repairNamespaces = false }
val repairingSerializer = XML { repairNamespaces = true }

// uses incorrect prefix "zdef-38325484" for property
println(repairingSerializer.stringify(PrefixBreaks.serializer(), thisExplodes))

// NPE, see below
println(defaultSerializer.stringify(PrefixBreaks.serializer(), thisExplodes))
```

The first `println` statement yields the following output:
```xml
<ex:namespaced xmlns:ex="http://example.org" xmlns:zdef-38325484="http://myownnamespace.com" zdef-38325484:property="bug"/>
```

The sourcecode explicitly specifies prefix `foo`, so the expected output would be:
```xml
<ex:namespaced xmlns:ex="http://example.org" xmlns:foo="http://myownnamespace.com" foo:property="bug"/>
```

The second `println` throws a `nl.adaptivity.xmlutil.XmlException` with the following stack trace:

```
Exception in thread "main" nl.adaptivity.xmlutil.XmlException: javax.xml.stream.XMLStreamException: Prefix cannot be null
	at nl.adaptivity.xmlutil.StAXWriter.doAttribute(StAXWriter.kt:245)
	at nl.adaptivity.xmlutil.StAXWriter.doFlushPending(StAXWriter.kt:147)
	at nl.adaptivity.xmlutil.StAXWriter.endTag(StAXWriter.kt:160)
	at nl.adaptivity.xmlutil.XmlWriterUtilCore.endTag(XmlWriter.kt:393)
	at nl.adaptivity.xmlutil.serialization.XmlEncoderBase$TagEncoder.endStructure(XMLEncoder.kt:270)
	at Playground$main$PrefixBreaks$$serializer.serialize(Playground.kt)
	at Playground$main$PrefixBreaks$$serializer.serialize(Playground.kt:63)
	at nl.adaptivity.xmlutil.serialization.XML.toXml(XML.kt:280)
	at nl.adaptivity.xmlutil.serialization.XML.stringify(XML.kt:189)
	at nl.adaptivity.xmlutil.serialization.XML.stringify(XML.kt:174)
	at Playground.main(Playground.kt:73)
Caused by: javax.xml.stream.XMLStreamException: Prefix cannot be null
	at java.xml/com.sun.xml.internal.stream.writers.XMLStreamWriterImpl.writeAttribute(XMLStreamWriterImpl.java:582)
	at nl.adaptivity.xmlutil.StAXWriter.doAttribute(StAXWriter.kt:242)
	... 10 more
```
where `Playground.kt:63` is the location of my `@Serializable` annotation on the `PrefixBreaks` class and `Playground.kt:73` is the call to `defaultSerializer.stringify`.

**Comment by suushiemaniac** [comment]
When using the Woodstox parser by adding `runtimeOnly("com.fasterxml.woodstox:woodstox-core:5.0.3")` to my Gradle dependencies, the behaviour changes as follows:

The first example essentially behaves the same, only that instead of the fantasy prefix `zdef-38325484` the fantasy namespace `wstxns1` gets used instead:

```xml
<ex:namespaced xmlns:ex="http://example.org" xmlns:wstxns1="http://myownnamespace.com" wstxns1:property="bug"/>
```

The second example still throws an exception, but this time with a different error and stacktrace:

```
Exception in thread "main" nl.adaptivity.xmlutil.XmlException: javax.xml.stream.XMLStreamException: Unbound namespace URI 'http://myownnamespace.com'
	at nl.adaptivity.xmlutil.StAXWriter.doAttribute(StAXWriter.kt:245)
	at nl.adaptivity.xmlutil.StAXWriter.doFlushPending(StAXWriter.kt:147)
	at nl.adaptivity.xmlutil.StAXWriter.endTag(StAXWriter.kt:160)
	at nl.adaptivity.xmlutil.XmlWriterUtilCore.endTag(XmlWriter.kt:393)
	at nl.adaptivity.xmlutil.serialization.XmlEncoderBase$TagEncoder.endStructure(XMLEncoder.kt:270)
	at Playground$main$PrefixBreaks$$serializer.serialize(Playground.kt)
	at Playground$main$PrefixBreaks$$serializer.serialize(Playground.kt:63)
	at nl.adaptivity.xmlutil.serialization.XML.toXml(XML.kt:280)
	at nl.adaptivity.xmlutil.serialization.XML.stringify(XML.kt:189)
	at nl.adaptivity.xmlutil.serialization.XML.stringify(XML.kt:174)
	at Playground.main(Playground.kt:73)
Caused by: javax.xml.stream.XMLStreamException: Unbound namespace URI 'http://myownnamespace.com'
	at com.ctc.wstx.sw.BaseStreamWriter.throwOutputError(BaseStreamWriter.java:1584)
	at com.ctc.wstx.sw.SimpleNsStreamWriter.writeAttribute(SimpleNsStreamWriter.java:77)
	at nl.adaptivity.xmlutil.StAXWriter.doAttribute(StAXWriter.kt:242)
	... 10 more
```

**Comment by pdvrieze** [comment]
Hello @suushiemaniac That's a very interesting one. The "fantasy" prefixes are consistent with a missing prefix. But that shouldn't happen (obviously). I'll have a look at what is going on here. It should use "foo" as prefix (although I don't think it handles duplicate prefixes in the same parent gracefully).

**Comment by pdvrieze** [comment]
I just updated it in master

**Comment by pdvrieze** [comment]
Now released in v0.20.0.10
